### PR TITLE
nrf54L dt fixes

### PR DESCRIPTION
**Pull request by SebastianBoe**
misc. nrf54L DT fixes

### CODE REVIEW

**Review by masz-nordic — DISMISSED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

**Review by kl-cruz — DISMISSED** [review]

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi [hunk 1]
```diff
@@ -60,7 +60,7 @@
 		sw1 = &button1;
 		sw2 = &button2;
 		sw3 = &button3;
-		watchdog0 = &wdt30;
+		watchdog0 = &wdt31;
```
Comment (rlubos) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi:63]: As we change the default watchdog instance here, we need to align the watchdog sample with this change, specifically those files:
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuflpr.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuflpr_xip.overlay
And test:
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/tests/drivers/watchdog/wdt_basic_api/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay

File: dts/common/nordic/nrf54l15.dtsi [hunk 2]
```diff
@@ -104,11 +104,14 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 
-		uicr: uicr@ffd000 {
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+/* intentionally empty because UICR is hardware fixed to Secure */
+#else
+uicr: uicr@ffd000 {
```
Comment (anangl) [inline dts/common/nordic/nrf54l15.dtsi:110]: The indentation could be improved here.

Comment (SebastianBoe) [reply dts/common/nordic/nrf54l15.dtsi]: done

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi (hunk 1 above)
Comment (SebastianBoe) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi:63]: done

**Review by anangl — DISMISSED** [review]

**Review by masz-nordic — DISMISSED** [review]

**Review by anangl — APPROVED** [review]
nit: the second commit title should start with "tests:", not "samples:".

**Review by rlubos — APPROVED** [review]

**Pull request merged**